tests/test_tileset.py

```diff
@@ -0,0 +1,243 @@
+import queue
+import datetime
+from unittest import TestCase, mock
+from typing import List, Dict, Any, Optional, Union, cast
+
+from tilekiln.tileset import Tileset
+from tilekiln.storage import Storage
+from tilekiln.config import Config
+from tilekiln.tile import Tile
+import tilekiln.errors
+
+
+class FakeStorage:
+    """
+    A fake implementation of Storage for testing purposes.
+    """
+    def __init__(self, calls: List[str], rets: queue.SimpleQueue) -> None:
+        self.calls = calls
+        self.rets = rets
+
+    def get_tileset(self, id: str) -> Any:
+        self.calls.append(f"get_tileset_{id}")
+        return self.rets.get()
+
+    def save_tile(
+        self,
+        id: str,
+        tile: Tile,
+        _layers: Dict[str, bytes],
+        _render_time: int = 0
+    ) -> Optional[Union[datetime.datetime, str]]:
```

Comment:
The signature of save_tile is `save_tile(self, tile: Tile, layers: dict[str, bytes])`, so the fake one should match